web/packages/teleport/src/Roles/RoleEditor/RoleEditor.tsx

```diff
@@ -87,6 +99,18 @@ export const RoleEditor = ({
     return roleToRoleEditorModel(parsedRole, originalRole?.object);
   });
 
+  const handlePreview = useCallback(async () => {
```

Comment:
Can we name it `handleYamlPreview` or somehow clarify in the comments why we are using both `useEffect` and `handlePreview` simultaneously? It's otherwise puzzling unless the person reading this is already familiar with the later parts of this function's body.